dvc/tree/gdrive.py

```diff
@@ -527,42 +543,47 @@ def exists(self, path_info, use_dvcignore=True):
         else:
             return True
 
-    def _list_paths(self, prefix=None):
-        if not self._ids_cache["ids"]:
-            return
-
-        if prefix:
-            dir_ids = self._ids_cache["dirs"].get(prefix[:2])
-            if not dir_ids:
-                return
-        else:
-            dir_ids = self._ids_cache["ids"]
-        parents_query = " or ".join(
-            f"'{dir_id}' in parents" for dir_id in dir_ids
-        )
-        query = f"({parents_query}) and trashed=false"
+    def isdir(self, path_info):
+        item_id = self._get_item_id(path_info)
+        file_repr = self._drive.CreateFile({"id": item_id})
+        self._fetch_metadata(file_repr, fields="mimeType")
+        return file_repr["mimeType"] == FOLDER_MIME_TYPE
 
-        for item in self._gdrive_list(query):
-            parent_id = item["parents"][0]["id"]
-            yield posixpath.join(
-                self._ids_cache["ids"][parent_id], item["title"]
-            )
+    def isfile(self, path_info):
+        return self.exists(path_info) and not self.isdir(path_info)
```

Comment:
Can we do it in one API call? For GDrive it is quite important, as there are pretty strict quotas.

Reply:
Done!